javascript/new-src/src/js/app.js

```diff
@@ -0,0 +1,497 @@
+/**
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+angular
+  .module('olaApp', ['ngRoute'])
+  .constant('Channels', 512)
+  .constant('TopValue', 255)
+  .constant('nul', 0)
+  .factory('$ola', ['$http', '$window', 'nul', 'Channels', function ($http, $window, nul, Channels) {
+    "use strict";
+    // once olad supports json post data postEncode can go away and the header in post requests too.
+    var postEncode = function (data) {
+      var PostData = [];
+      for (var key in data) {
+        PostData.push(key + '=' + data[key]);
+      }
+      return PostData.join('&');
+    };
+    var dmxConvert = function (dmx) {
+      var length = 1, integers = [];
+      for (var i = 0; i < Channels; i++) {
+        integers[i] = parseInt(dmx[i], 10);
+        if (integers[i] > nul) {
```

Comment:
since it loops trough 0 to 512, no. it remembers the highest position of the non zero value, the rest are all zeros so they will just get thrown away.